### PR TITLE
Updated schema, api for enhanced calendar functionality

### LINKED ISSUE #53: Add separate teacher and student zoom meeting links for classes
Teachers need to use the 'start_url' which is what tells the meeting to begin
Students should only be using the 'join_url' which waits for the teacher to join

### PR DESCRIPTION
**Pull request by jdurelli**

### CODE REVIEW

**Comment by jdurelli** [comment]
Does this close any other issues? Should we add an issue for the general class schedule functionality changes?

**Review by jonathanmcdaniel — APPROVED** [review]

File: RecessApplication/api.py [hunk 1]
```diff
@@ -56,33 +57,69 @@ def getLogger(self):
...
+
+        if not self.exists(enrollments):
             return Response({
                 "error": "User is not enrolled in or teaching any classes."
             })
         
-        class_ids = [ e['class_id'] for e in enrollments ]
-        classes = Class.objects.filter(class_id__in=class_ids).values()
-        class_schedules = ClassSchedule.objects.filter(class_id__in=class_ids).values()
+        year = request.data['year']
```
Comment (jonathanmcdaniel) [inline RecessApplication/api.py]: We need to change this to be a url parameter not  body data

File: RecessApplication/api.py [hunk 2]
```diff
@@ -56,33 +57,69 @@ def getLogger(self):
...
+        week = request.data['week']
```
Comment (jonathanmcdaniel) [inline RecessApplication/api.py]: we need to change this to be a url parameter not body data

**Review by jonathanmcdaniel — CHANGES_REQUESTED** [review]
Because we are working off of a GET request, standard practice is to not have a body in a GET method because you are supposed to be requesting data, not sending it. I didnt think this was going to be an issue since we saw it work from postman, but react will not allow this request to be made.

**Review by jonathanmcdaniel — APPROVED** [review]

**Review by gateship1 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**